handlers/main.yml

```diff
@@ -1,18 +1,15 @@
 ---
-# handlers file for tinc
-
-#- name: Restart Service
-#  service:
-#    name: "{{ service_name }}"
-#    daemon_reload: "{{ ansible_service_mgr == 'systemd' | ternary('yes',omit) }}"
-#    state: restarted
+- name: restart service
+  service:
+    name: "{{ tinc_service_name }}"
+    state: restarted
+    enabled: yes
+  when: ansible_service_mgr != "systemd"
 
-- name: Restart Service
+- name: restart service with systemd
   systemd:
    daemon_reload: yes
+   name: "{{ tinc_service_name }}"
+   state: restarted
+   enabled: yes
   when: ansible_service_mgr == "systemd"
-
```

Comment:
Side note for the context: This was created when a bug to the module "service" wouldn't allow to restart the service, enable it, and daemon_reload at the same time.

As you're cleaning that up (I suppose that bug is fixed:) ), I think we can come back to the initial "Restart Service" handler. If we change the handler name, it would basically change the user interface: the handlers is something exposed to the users of the role. In order to not change the user interface, could you keep the words "Restart Service" for the handler name?

Also, I am not sure we need two tasks if we clean this up, the ternary operator to omit daemon_reload should be good enough, what do you think?

Next to all of this, I think it would be nice if it was done in a separate commit. This way we keep the history clean of what does each of those: One is tackling molecule, one is changing the service (in case we ever need to revert, it's easier to have a clean scope)